R/geotargets-option.R

```diff
@@ -0,0 +1,50 @@
+#' Get or Set geotargets Options
+#'
+#' Get or set behavior for geospatial data target stores using geotargets-specific global options.
+#'
+#' @param x Character. Option name. See Details.
+#'
+#' @details
+#'
+#' ## Available Options
+#'
+#'  - `"geotargets.raster.gdal_creation_options"` - set the GDAL creation options used when writing raster files to target store (default: `"ENCODING=UTF-8"`)
+#'
+#'  - `"geotargets.raster.gdal_driver_name"` - set the file type used for raster data in target store (default: `"GTiff"`)
+#'
+#'  Each option can be overridden with a system environment variable. Options include:
+#'
+#'   - `GEOTARGETS_RASTER_GDAL_CREATION_OPTIONS`
+#'   - `GEOTARGETS_RASTER_GDAL_DRIVER_NAME`
+#'
+#' @rdname geotargets-options
+#' @export
+geotargets_option_get <- function(x) {
+    if (!startsWith(x, "geotargets.")) {
+        x <- paste0("geotargets.", x)
+    }
+
+    value <- geotargets_env()[[x]]
+
+    switch(x,
+           "geotargets.raster.gdal_creation_options" = {
+               strsplit(Sys.getenv("GEOTARGETS_RASTER_GDAL_CREATION_OPTIONS",
+                                   unset = getOption(x, default = ifelse(is.null(value), "ENCODING=UTF-8", value))),
+                        ";")[[1]]
+           },
+           "geotargets.raster.gdal_driver_name" = {
+               Sys.getenv("GEOTARGETS_RASTER_GDAL_DRIVER_NAME",
+                          unset = getOption(x, default = ifelse(is.null(value), "GTiff", value))
+               )
+           })
+}
```

Comment:
Naming things is hard: some alternatives for these functions:

- get_geotargets_raster_gdal_creation_options
  - get_geotargets_gdal_options_raster_creation
  - get_geotargets_gdal_raster_options_creation

- get_geotargets_raster_gdal_driver_name
  - get_geotargets_gdal_name_raster_driver
  - get_geotargets_gdal_raster_name_driver

Thoughts welcome! I also appreciate that this is a small detail, so perhaps it isn't that important

Reply:
Totally fine with an rlang import, but you'll have to forgive me when I don't think to use it while I am prototyping something. Good suggestion there and makes things clearer.

I also thought about putting `"_gdal_"` before the geospatial type in the option name, I would be fine with that. 

Breaking out these functions is a little more abstraction than I normally would do for something like this, but it may be good to have explicit setters for each option to facilitate setting them in a clear way. I ultimately don't have a problem with defining these specific functions other than we might wind up with a lot of them if we continue on this path

I imagine we could have analogous options for vector data, so the "raster" bit should come relatively early in the name so the user "branches early" for autocomplete etc. 

 I think I prefer either `get_geotargets_<raster/vector>_gdal_...()` or `get_geotargets_gdal_<raster/vector>_...()`

With "creation options" and "driver name", I am referring to GDAL constructs for these things (i.e. https://gdal.org/drivers/raster/gtiff.html#creation-options, https://gdal.org/drivers/raster/index.html). Trying to make it generic, not specifically based off of the argument names for terra functions. For example the sf analog of `terra::writeVector(..., filetype="foo", gdal="bar")` is `sf::st_write(..., driver="foo", layer_options="bar")`

Reply:
I think I like the suggestion of putting "gdal" before "raster" or "vector"--it seems to be easier to read to me.